file_lib/src/checksum.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use digest::Digest;
+use sha1::Sha1;
+use sha2::{Sha256, Sha512};
+use hex_string::HexString;
+
+/// The supported checksum hash algorithms.
+pub enum Algorithm {
+    /// The SHA-1 algorithm. (Considered insecure.)
+    Sha1,
+    /// The SHA-256 algorithm.
+    Sha256,
+    /// The SHA-512 algorithm.
+    Sha512,
+}
+
+/// Compute the checksum for a vector of bytes using a given algorithm.
```

Comment:
💜❗

These reference docs are great, doubly love the inclusion of examples-as-tests! This is a pretty straightforward function but this information and reference documentation will make it much easier to write the higher level docs when we get to the resource implementation. 😊